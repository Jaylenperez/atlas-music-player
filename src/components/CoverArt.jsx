import cover from "../assets/placeholder.svg";

export function CoverArt() {
  return (
    <div className="pb-5">
      <img src={cover} alt="Cover Art" className="size-100" />
    </div>
  );
}
