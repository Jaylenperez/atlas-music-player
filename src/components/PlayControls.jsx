export function PlayControls() {
  return (
    <div>
      <button>
        <svg
          xmlns="http://www.w3.org/2000/svg"
          width="24"
          height="24"
          viewBox="0 0 24 24"
          fill="none"
          stroke="currentColor"
          stroke-width="2"
          stroke-linecap="round"
          stroke-linejoin="round"
          class="lucide lucide-rewind-icon lucide-rewind"
        >
          <polygon points="11 19 2 12 11 5 11 19" />
          <polygon points="22 19 13 12 22 5 22 19" />
        </svg>
      </button>
      <button>
        <svg
          xmlns="http://www.w3.org/2000/svg"
          width="24"
          height="24"
          viewBox="0 0 24 24"
          fill="none"
          stroke="currentColor"
          stroke-width="2"
          stroke-linecap="round"
          stroke-linejoin="round"
          class="lucide lucide-play-icon lucide-play"
        >
          <polygon points="6 3 20 12 6 21 6 3" />
        </svg>
      </button>
      <button>
        <svg
          xmlns="http://www.w3.org/2000/svg"
          width="24"
          height="24"
          viewBox="0 0 24 24"
          fill="none"
          stroke="currentColor"
          stroke-width="2"
          stroke-linecap="round"
          stroke-linejoin="round"
          class="lucide lucide-fast-forward-icon lucide-fast-forward"
        >
          <polygon points="13 19 22 12 13 5 13 19" />
          <polygon points="2 19 11 12 2 5 2 19" />
        </svg>
      </button>
      <button>
        <svg
          xmlns="http://www.w3.org/2000/svg"
          width="24"
          height="24"
          viewBox="0 0 24 24"
          fill="none"
          stroke="currentColor"
          stroke-width="2"
          stroke-linecap="round"
          stroke-linejoin="round"
          class="lucide lucide-shuffle-icon lucide-shuffle"
        >
          <path d="m18 14 4 4-4 4" />
          <path d="m18 2 4 4-4 4" />
          <path d="M2 18h1.973a4 4 0 0 0 3.3-1.7l5.454-8.6a4 4 0 0 1 3.3-1.7H22" />
          <path d="M2 6h1.972a4 4 0 0 1 3.6 2.2" />
          <path d="M22 18h-6.041a4 4 0 0 1-3.3-1.8l-.359-.45" />
        </svg>
      </button>
    </div>
  );
}
