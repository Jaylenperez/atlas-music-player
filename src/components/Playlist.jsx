// src/components/Playlist.jsx
import React, { useState } from "react";

const tracks = [
  { title: "Painted in Blue", artist: "Soul Canvas", length: "5:55" },
  { title: "Tidal Drift", artist: "Echoes of the Sea", length: "8:02" },
  { title: "Fading Shadows", artist: "The Emberlight", length: "3:01" },
  { title: "Cosmic Drift", artist: "Solar Flare", length: "5:01" },
  { title: "Urban Serenade", artist: "Midnight Groove", length: "4:54" },
  { title: "Whispers in the Wind", artist: "Rust & Ruin", length: "6:13" },
  { title: "Electric Fever", artist: "Neon Jungle", length: "8:41" },
  { title: "Edge of the Abyss", artist: "Steel Horizon", length: "2:27" },
  { title: "Golden Haze", artist: "Velvet Waves", length: "3:15" },
  { title: "Shatter the Silence", artist: "Thunderclap Echo", length: "8:22" },
];

export default function Playlist({ lightMode }) {
  const [selectedIndex, setSelectedIndex] = useState(0);

  return (
    <div className="w-full">
      {/* Playlist heading */}
      <h2
        className={`mb-4 text-left text-2xl font-semibold ${
          lightMode ? "text-slate-950" : "text-sky-300"
        }`}
      >
        Playlist
      </h2>

      {/* Track list */}
      <div className="space-y-1">
        {tracks.map((track, idx) => (
          <div
            key={idx}
            onClick={() => setSelectedIndex(idx)}
            className={`cursor-pointer rounded-md border-b ${lightMode ? "border-stone-600" : "border-white"} /* light=brown, dark=white */ px-4 py-2 transition ${
              idx === selectedIndex
                ? lightMode
                  ? "bg-gray-200"
                  : "bg-indigo-900"
                : ""
            } ${lightMode ? "hover:bg-gray-200" : "hover:bg-indigo-900"} `}
          >
            <div className="flex items-center justify-between">
              <div>
                {/* Song title */}
                <h3
                  className={`text-lg font-medium ${
                    lightMode ? "text-gray-900" : "text-sky-400"
                  }`}
                >
                  {track.title}
                </h3>
                {/* Artist */}
                <p
                  className={`text-sm ${
                    lightMode ? "text-rose-800" : "text-fuchsia-500"
                  }`}
                >
                  {track.artist}
                </p>
              </div>
              {/* Length */}
              <span
                className={`text-sm ${
                  lightMode ? "text-gray-600" : "text-fuchsia-300"
                }`}
              >
                {track.length}
              </span>
            </div>
          </div>
        ))}
      </div>
    </div>
  );
}
