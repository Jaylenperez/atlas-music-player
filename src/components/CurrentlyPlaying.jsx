// src/components/CurrentlyPlaying.jsx
import { CoverArt } from "./CoverArt";
import { SongTitle } from "./SongTitle";
import { PlayControls } from "./PlayControls";
import { VolumeControls } from "./VolumeControls";

export default function CurrentlyPlaying({ lightMode }) {
  return (
    <div className="flex h-full w-full flex-col justify-between px-4 py-4">
      {/* Top: cover art */}
      <div className="flex-shrink-0">
        <CoverArt />
      </div>

      {/* Middle: song title + play/pause/etc */}
      <div className="space-y-3">
        <SongTitle lightMode={lightMode} />
        <PlayControls />
      </div>

      {/* Bottom: volume slider */}
      <div className="flex-shrink-0">
        <VolumeControls />
      </div>
    </div>
  );
}
