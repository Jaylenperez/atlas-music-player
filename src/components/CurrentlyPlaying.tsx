// src/components/CurrentlyPlaying.tsx
import React, { useEffect, useState } from "react";
import { CoverArt } from "./CoverArt";
import { SongTitle } from "./SongTitle";
import { PlayControls } from "./PlayControls";
import { VolumeControls } from "./VolumeControls";

interface PlaylistSong {
  id: string;
  title: string;
  artist: string;
  genre: string;
  duration: number;
}

interface Song extends PlaylistSong {
  cover: string;
  song: string;
}

interface CurrentlyPlayingProps {
  lightMode: boolean;
  track: PlaylistSong;
}

export const CurrentlyPlaying: React.FC<CurrentlyPlayingProps> = ({
  lightMode,
  track,
}) => {
  // we'll store the full Song here once we fetch it
  const [fullSong, setFullSong] = useState<Song | null>(null);

  useEffect(() => {
    // every time `track.id` changes, fetch its details
    fetch(`/api/v1/songs/${track.id}`)
      .then((res) => {
        if (!res.ok) throw new Error("Failed to load song details");
        return res.json();
      })
      .then((data: Song) => setFullSong(data))
      .catch((err) => {
        console.error(err);
        setFullSong(null);
      });
  }, [track.id]);

  // while loading, you can show a spinner or fallback image
  const coverSrc = fullSong?.cover ?? "/placeholder.svg";

  return (
    <div className="flex h-full w-full flex-col justify-between px-4 py-4">
      <div className="flex-shrink-0">
        <CoverArt src={coverSrc} />
      </div>

      {fullSong && (
        <div className="space-y-3">
          <SongTitle
            lightMode={lightMode}
            title={fullSong.title}
            artist={fullSong.artist}
          />
          <PlayControls lightMode={lightMode} />
        </div>
      )}

      <div className="flex-shrink-0">
        <VolumeControls lightMode={lightMode} />
      </div>
    </div>
  );
};
