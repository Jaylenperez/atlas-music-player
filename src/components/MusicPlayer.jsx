import { CoverArt } from "./CoverArt";
import { SongTtile } from "./SongTitle";
import { PlayControls } from "./PlayControls";
import { VolumeControls } from "./VolumeControls";
import PlayListItem from "./PlayListItem";

export default function MusicPlayer() {
  return (
    <div>
      <CoverArt />
      <SongTtile />
      <PlayControls />
      <VolumeControls />
      <PlayListItem
        title="Nightmare"
        artist="Avenged Sevenfold"
        length="6:14"
      />
    </div>
  );
}
