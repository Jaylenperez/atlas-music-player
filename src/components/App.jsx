// src/App.jsx
import React, { useState } from "react";
import CurrentlyPlaying from "./CurrentlyPlaying";
import Playlist from "./Playlist";

export default function App() {
  const [lightMode, setLightMode] = useState(false);

  return (
    <div
      className={`${
        lightMode ? "bg-gray-100 text-gray-900" : "bg-indigo-950 text-white"
      } flex h-screen flex-col overflow-x-hidden md:flex-row`}
    >
      {/* toggle button */}
      <button
        onClick={() => setLightMode(!lightMode)}
        className={`absolute top-4 right-4 rounded px-3 py-1 text-sm font-medium ${
          lightMode ? "bg-slate-400 text-gray-800" : "bg-gray-800 text-gray-200"
        } transition`}
      >
        {lightMode ? "Dark Mode" : "Light Mode"}
      </button>

      {/* left panel */}
      <div className="flex w-full flex-col justify-between p-4 md:w-1/2 md:p-8">
        <CurrentlyPlaying />
      </div>

      {/* right panel */}
      <div className="w-full overflow-y-auto border-t border-gray-200 p-4 md:w-1/2 md:border-t-0 md:border-l md:p-8">
        <Playlist lightMode={lightMode} />
      </div>
    </div>
  );
}
