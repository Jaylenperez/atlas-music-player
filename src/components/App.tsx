import React, { useState, useEffect } from "react";
import { CurrentlyPlaying } from "./CurrentlyPlaying";
import Playlist from "./Playlist";

export default function App() {
  const [lightMode, setLightMode] = useState(false);

  // ← new state for the full playlist and the selected item
  const [playlist, setPlaylist] = useState([]);
  const [selectedTrack, setSelectedTrack] = useState(null);

  useEffect(() => {
    fetch("/api/v1/playlist")
      .then((res) => res.json())
      .then((data) => {
        setPlaylist(data);
        if (data.length > 0) setSelectedTrack(data[0]);
      })
      .catch((err) => console.error("Failed to load playlist:", err));
  }, []);

  return (
    <div
      className={`${
        lightMode ? "bg-red-50 text-gray-900" : "bg-indigo-950 text-white"
      } flex h-screen flex-col overflow-x-hidden md:flex-row`}
    >
      <button
        onClick={() => setLightMode(!lightMode)}
        className={`absolute top-4 right-4 rounded px-3 py-1 text-sm font-medium ${
          lightMode ? "bg-slate-400 text-gray-800" : "bg-gray-800 text-gray-200"
        } transition`}
      >
        {lightMode ? "Dark Mode" : "Light Mode"}
      </button>

      <div className="flex w-full flex-col justify-between p-4 md:w-1/2 md:p-8">
        {/* pass the entire track object (including cover, duration, etc.) */}
        {selectedTrack && (
          <CurrentlyPlaying lightMode={lightMode} track={selectedTrack} />
        )}
      </div>

      <div className="w-full overflow-y-auto border-t border-gray-200 p-4 md:w-1/2 md:border-t-0 md:border-l md:p-8">
        <Playlist
          lightMode={lightMode}
          tracks={playlist}
          selected={selectedTrack}
          onSelect={setSelectedTrack}
        />
      </div>
    </div>
  );
}
