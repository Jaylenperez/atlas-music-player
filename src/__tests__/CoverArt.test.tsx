// src/__tests__/CoverArt.test.tsx
import React from "react";
import { render, screen } from "@testing-library/react";
import { describe, test, expect } from "vitest";
import { CoverArt } from "../components/CoverArt";

describe("CoverArt component", () => {
  test("matches snapshot with default alt", () => {
    const { container } = render(
      <CoverArt src="http://example.com/cover.jpg" />
    );
    expect(container).toMatchSnapshot();
  });

  test("matches snapshot with custom alt", () => {
    const { container } = render(
      <CoverArt src="http://example.com/cover.jpg" alt="Album Cover" />
    );
    expect(container).toMatchSnapshot();
  });

  test("renders image with correct src and alt attributes", () => {
    render(<CoverArt src="http://example.com/cover2.png" />);
    const img = screen.getByRole("img", { name: /cover art/i });
    expect(img).toBeInTheDocument();
    expect(img).toHaveAttribute("src", "http://example.com/cover2.png");
    expect(img).toHaveAttribute("alt", "Cover Art");
  });

  test("applies correct Tailwind classes to wrapper and img", () => {
    render(<CoverArt src="dummy-url" alt="Test Alt" />);
    const img = screen.getByRole("img", { name: /test alt/i });
    const wrapper = img.parentElement;

    // wrapper should have flex & justify classes
    expect(wrapper).toHaveClass("flex", "justify-center", "md:justify-start");

    // img should have responsive aspect-square and width classes
    expect(img).toHaveClass(
      "aspect-square",
      "w-full",
      "rounded-md",
      "object-cover",
      "sm:w-72",
      "md:w-80",
      "lg:w-96",
      "xl:w-[50rem]"
    );
  });
});
