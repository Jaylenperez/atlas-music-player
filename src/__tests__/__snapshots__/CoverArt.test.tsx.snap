// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CoverArt component > matches snapshot with custom alt 1`] = `
<div>
  <div
    class="flex justify-center md:justify-start"
  >
    <img
      alt="Album Cover"
      class="aspect-square w-full rounded-md object-cover sm:w-72 md:w-80 lg:w-96 xl:w-[50rem]"
      src="http://example.com/cover.jpg"
    />
  </div>
</div>
`;

exports[`CoverArt component > matches snapshot with default alt 1`] = `
<div>
  <div
    class="flex justify-center md:justify-start"
  >
    <img
      alt="Cover Art"
      class="aspect-square w-full rounded-md object-cover sm:w-72 md:w-80 lg:w-96 xl:w-[50rem]"
      src="http://example.com/cover.jpg"
    />
  </div>
</div>
`;
